app/src/main/java/org/wildstang/wildrank/androidv2/activities/SyncActivity.java

```diff
@@ -50,10 +60,37 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     private void beginSync() {
-        if (!SyncUtilities.isFlashDriveConnected()) {
-            showDriveNotMountedWarning();
+        if (SyncUtilities.useSAF()) {
+            SyncUtilities.openFileChooser(this);
         } else {
-            new SyncTask().execute();
+            if (!SyncUtilities.isFlashDriveConnected(getApplicationContext())) {
+                showDriveNotMountedWarning();
+            } else {
+                new SyncTask().execute();
+            }
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == SyncUtilities.REQUEST_CODE_OPEN && resultCode == Activity.RESULT_OK) {
```

Comment:
I see this is duplicated in both the setup and sync activities, you should be able to wrap this up into a static helper function in sync utilities that takes a context and intent?